test/functional/feature_reindex.py

```diff
@@ -68,12 +72,32 @@ def find_block(b, start):
             'LoadExternalBlockFile: Out of order block',
             'LoadExternalBlockFile: Processing out of order child',
         ]):
-            extra_args = [["-reindex"]]
-            self.start_nodes(extra_args)
+            self.start_node(0, ["-reindex"])
 
         # All blocks should be accepted and processed.
         assert_equal(self.nodes[0].getblockcount(), 12)
 
+    def reindex_readonly(self):
+        self.connect_nodes(0, 1)
+        # generate enough blocks to ensure that the -fastprune node fills up the
+        # first blk00000.dat file and starts another block file
+        self.generatetoaddress(self.nodes[1], 300, self.nodes[1].get_deterministic_priv_key().address)
+        self.stop_node(1)
+
+        # make the first block file read-only
+        path = os.path.join(self.nodes[1].chain_path, 'blocks', 'blk00000.dat')
+        os.chmod(path, 0o444)
+
+        # restart and reindex the node with the read-only block file
+        self.start_node(1, ["-reindex", "-fastprune"])
+        assert_equal(self.nodes[1].getblockcount(), 312)
+
+        # shut down this node, triggering disk flushes, but not to the read-only file
+        with self.nodes[1].assert_debug_log(expected_msgs=['Flush'], unexpected_msgs=['Error']):
+            self.stop_node(1)
```

Comment:
There are many `Flush` log lines for unrelated things (like the ChainState), so consider making that string more specific. Maybe:
```
2023-02-04T00:15:05.877123Z [shutoff] [logging/timer.h:57] [Log] [bench] FlushStateToDisk: write block and undo data to disk completed (6.62ms)
```
Similarly, maybe "Error" is too general? It does work, but I'd be afraid that something unrelated might log "Error" in the future and break this test. Maybe look for one of the error strings in my comment above?

Reply:
good call, I was being too general, updated. thanks!